src/orion/benchmark/task/quadratics.py

```diff
@@ -0,0 +1,113 @@
+""" Defines the Quadratics task, as described in section 4.2 of the ABLR paper.
+
+[1] [Scalable HyperParameter Transfer Learning](
+    https://papers.nips.cc/paper/7917-scalable-hyperparameter-transfer-learning)
+"""
+from dataclasses import dataclass
+from logging import getLogger as get_logger
+from typing import Dict, Optional, List
+
+import numpy as np
+from orion.benchmark.task.base import BenchmarkTask
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class QuadraticsTaskHParams:
+    """ Hyper-parameters of the Quadratics task. """
+
+
+@dataclass
+class QuadraticsTaskHparamsWithContext(QuadraticsTaskHParams):
+    """ Hyper-parameters of the Quadratics task that also include the context vector.
+
+    The context vector in this case are the coefficients of the quadratic.
+    
+    NOTE: This adds these entries to the space, but they are ignored when evaluating a given point
+    (they aren't used by the `__call__` method of QuadraticsTask).
+    """
+
+    a2: float
+    a1: float
+    a0: float
+
+
+class QuadraticsTask(BenchmarkTask):
+    """ Simple task consisting of a quadratic with three coefficients, as described in ABLR.
+    
+   Notes
+   -----
+   In the paper, section 4.2, the function is defined over R^3, but here
+   the bounds are limited to [-100,100] for now since otherwise y can be enormous.
+
+    Parameters
+    ----------
+    max_trials : int
+        Maximum number of tr
+    a2 : float, optional
+        a2 coefficient, by default None, in which case it is sampled in the [0.1, 10.0] interval.
+    a1 : float, optional
+        a1 coefficient, by default None, in which case it is sampled in the [0.1, 10.0] interval.
+    a0 : float, optional
+        a0 coefficient, by default None, in which case it is sampled in the [0.1, 10.0] interval.
+    seed : int, optional
+        Random seed, by default None
+    with_context : bool, optional
+        Wether to append the values of (a2, a1, a0) to the points sampled from this task. 
+    """
+
+    def __init__(
+        self,
+        max_trials: int,
+        a2: float = None,
+        a1: float = None,
+        a0: float = None,
+        seed: int = None,
+        with_context: bool = False,
+    ):
+        super().__init__(max_trials=max_trials)
+        self.seed: Optional[int] = seed
+        self.rng = np.random.default_rng(self.seed)
+        # Note: rounding to 4 decimals just to prevent some potential issues
```

Comment:
This should be specified in documentation instead.

Reply:
Correct, fixed in bafe8b19219d31d292051f88b6b821cd388429d6